### PR TITLE
[gui] do not  automatically format name of added layers (fixes #17741)

### PR DESCRIPTION
**Pull request by nirvn**
## Description
@nyalldawson , this partially reverts a previous commit of yours that enforced auto capitalization and underscore-to-space formatting of the name of newly added layers. This address a few complaints raised on issues.qgis.org.

If users / organizations / companies are in need of auto formatting, a settings variable (qgis/formatLayerName) can be switched to true.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: resources/qgis_global_settings.ini [hunk 1]
```diff
@@ -1,8 +1,9 @@
 [qgis]
+formatLayerName=false
```
Comment (nyalldawson) [inline resources/qgis_global_settings.ini]: Can we add comments to this file? It'd certainly help organisations customise their own version of this...

File: src/app/qgsrasterlayerproperties.cpp [hunk 2]
```diff
@@ -1066,7 +1066,12 @@ void QgsRasterLayerProperties::apply()
 
 void QgsRasterLayerProperties::mLayerOrigNameLineEd_textEdited( const QString &text )
 {
-  leDisplayName->setText( mRasterLayer->formatLayerName( text ) );
+  QgsSettings settings;
```
Comment (nyalldawson) [inline src/app/qgsrasterlayerproperties.cpp]: I think this one is safe to leave. What do you think?

File: src/app/qgsvectorlayerproperties.cpp [hunk 3]
```diff
@@ -843,7 +843,12 @@ QString QgsVectorLayerProperties::htmlMetadata()
 
 void QgsVectorLayerProperties::mLayerOrigNameLineEdit_textEdited( const QString &text )
 {
-  txtDisplayName->setText( mLayer->formatLayerName( text ) );
+  QgsSettings settings;
```
Comment (nyalldawson) [inline src/app/qgsvectorlayerproperties.cpp]: ditto

File: src/app/qgsrasterlayerproperties.cpp (hunk 2 above)
Comment (nirvn) [reply src/app/qgsrasterlayerproperties.cpp]: @nyalldawson , what do you mean? Remove the formatting altogether?

**Comment by nirvn** [comment]
@nyalldawson , comments addressed, thanks.

**Pull request merged**